Add details for redirecting constructors (#3426)
dart-lang/site-www: src/_guides/language/language-tour.md

Sometimes a constructor’s only purpose is to redirect to another
constructor in the same class. A redirecting constructor’s body is

> empty, with the constructor call appearing after a colon (:). The
> constructor being redirected to must be accessed using the `this`
> keyword.

Reviewer: Yep, our laconic style is perhaps a bit too terse here. Since there are multiple uses of `this` in the following example, how about something like (wait for it) this:

```suggestion
empty, with the constructor call
(using `this` instead of the class name)
appearing after a colon (:).
```

We could also highlight `: this(x, 0);`. I can figure out the code for that if you think it's a good idea.

Author: Thanks I like your wording. I don't think we necessarily need to highlight that code.
